source/fundamentals/builders/filters.txt

```diff
@@ -11,3 +11,684 @@ Filters Builders
+
+   * - ``regex()``
+     - Selects documents where values match a specified regular expression.
+      
+   * - ``text()``
+     - Selects documemts matching a full-text search expression.
+     
+   * - ``where()``
+     - Matches documents that satisfy a JavaScript expression.
+
+The following code snippet shows how to use an evaluation filter:
+
+.. code-block:: java
+   
+   import static com.mongodb.client.model.Filters.*;
+
+   // <MongoCollection setup code here>
+
+   collection.find(<evaluation filter>);
+
+.. example::
+
+   The following example selects documents that have a ``color`` field
+   containing the term ``green`` in the paint_purchaces collection:
+
+   .. code-block:: java
+      
+      import static com.mongodb.client.model.Filters.text;
+
+      // <MongoCollection setup code here>
+
+      Bson textComparison = text("green");
+      List<Document> results = new ArrayList<>();
+      collection.find(textComparison).into(results);
+      for (Document result : results) {
+         System.out.println(result.toJson());
+      }
+
+   The Bson renders as:
+
+   .. code-block:: none
+      :copyable: false
+   
+      { 
+         "text" : "green"
+      }
+
+   The output of the above code snippet should look something like this:
+   
+   .. code-block:: json
+      :copyable: false
+      
+      {"_id": 7, "color": "green", "qty": 8,"vendor": ["B", "C"]}
+
+   Available with MongoDB 3.2, a version 3 text index allows case-sensitive
+   and diacritic-sensitive searches.
+
+   .. code-block:: java
+
+      text("green", TextSearchOptions().caseSensitive(true))
+
+   .. code-block:: java
+
+      text("green", TextSearchOptions().diacriticSensitive(true))
+
+.. _bitwise:
+
+Bitwise
+-------
+
+The bitwise query operators include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Logical Operator
+     - Selects documents where
+      
+   * - ``bitsAllSet()``
+     - the specified bits of a field are set (i.e. 1).
+
+   * - ``bitsAllClear()``
+     - the specified bits of a field are clear (i.e. 0).
+      
+   * - ``bitsAnySet()``
+     - at least one of the specified bits of a field are set (i.e. 1).
+     
+   * - ``bitsAnyClear()``
+     - at least one of the specified bits of a field are clear (i.e. 0).
+
+The following code snippet shows how to use a bitwise filter:
+
+.. code-block:: java
+   
+   import static com.mongodb.client.model.Filters.*;
+
+   // <MongoCollection setup code here>
+
+   collection.find(<bitwise filter>);
+
+.. example::
+
+   The following example selects documents that have a bitField field
+   with bits set at positions of the corresponding bitmask 50 (i.e.
+   00110010) in the binary_numbers collection: 
+
+   .. code-block:: java
+      
+      import static com.mongodb.client.model.Filters.bitsAllSet;
+
+      // <MongoCollection setup code here>
+
+      Bson bitsComparison = bitsAllSet(1, 5);
```

Comment:
I think there might be some confusion here in what parameters you provide this method. For example, how do we know this is for field "a"? And isn't a bitmask of 5 equivalent to 0x101 which doesn't match the results in the example? 

Could you check the API documentation and correct it? Also, it could be useful to mention something about the parameters in the descriptions in the table above.

Reply:
should I add a column in each table for the parameters of each method?

Reply:
It might be useful at some point to link these to API documentation, but can you explain examples that might not be self-explanatory like this one? From this API documentation page, unless I'm missing something, I'm not sure this will compile since the first argument should be a String:
https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Filters.html#bitsAllSet(java.lang.String,long)